### ISSUE TITLE
DBusException: Cannot create an item in a locked collection

### ISSUE THREAD

**Issue by jaraco** [jaraco/keyring#69]
```
#!python
      File "/usr/lib64/python2.7/site-packages/keyring/core.py", line 42, in set_password
        _keyring_backend.set_password(service_name, username, password)
      File "/usr/lib64/python2.7/site-packages/keyring/backend.py", line 233, in set_password
            True)
      File "/usr/lib64/python2.7/site-packages/dbus/proxies.py", line 70, in __call__
    return self._proxy_method(*args, **keywords)
      File "/usr/lib64/python2.7/site-packages/dbus/proxies.py", line 145, in __call__
        **keywords)
      File "/usr/lib64/python2.7/site-packages/dbus/connection.py", line 651, in call_blocking
        message, timeout)
    dbus.exceptions.DBusException: org.freedesktop.Secret.Error.IsLocked: Cannot create an item in a locked collection
```

---
- Bitbucket: https://bitbucket.org/kang/python-keyring-lib/issue/69
- Originally reported by: Anonymous
- Originally created at: 2012-08-03T00:46:39.141

**Comment by jaraco** [comment]
I apologize, but I can't do much with just this error message. Can you clarify with some context and detail:

# What were you doing when the error occurred? What steps can someone else take to recreate the error?

# What did you expect to happen?

# What was your operating system and Python version?

# What other detail can you provide that might hint at the cause of the problem?

I'm going to close this ticket as wontfix, only because there's little that can be done without more information, but don't let that discourage you. Just provide more information and re-open the ticket.

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
Hi,

In Gnome 3.6, when the key ring is locked:
- `get_password()` always returns `None`, even if the key ring _has_ a matching password,
- `set_password()` raises the exception shown above.

If the key ring is then unlocked through another application (eg. Empathy), everything starts working as expected.

Expected result: when the key ring is locked, the `keyring` module should have Gnome show the usual password prompt to unlock it.

---

Original comment by: Simon Sapin

**Comment by jaraco** [comment]
Same here (openSUSE, keyring-1.2.2). 

---

Original comment by: Sascha Peilicke

**Comment by jaraco** [comment]
This issue is fixed in my pull request #25.

---

Original comment by: Dmitry Shachnev
